DashAI/back/models/hugging_face/opus_mt_en_es_transformer.py

```diff
@@ -0,0 +1,151 @@
+        ----------
+        dataset : DashAIDataset
+            DashAIDataset with training data.
+
+        """
+
+        input_column = dataset.inputs_columns[0]
+        output_column = dataset.outputs_columns[0]
+
+        def tokenize(examples):
+            inputs = self.tokenizer(
+                examples[input_column],
+                truncation=True,
+                padding="max_length",
+                max_length=512,
+            )
+            outputs = self.tokenizer(
+                examples[output_column],
+                truncation=True,
+                padding="max_length",
+                max_length=512,
+            )
+            inputs["labels"] = outputs["input_ids"]
+            return inputs
+
+        dataset = dataset.map(tokenize, batched=True)
+        dataset.set_format("torch", columns=["input_ids", "attention_mask", "labels"])
+
+        # Arguments for fine-tuning
+        training_args = Seq2SeqTrainingArguments(
+            output_dir="DashAI/back/user_models/temp_checkpoints_opus-mt-en-es",
+            save_steps=1,
+            save_total_limit=1,
+            per_device_train_batch_size=self.batch_size,
+            per_device_eval_batch_size=self.batch_size,
+            no_cuda=self.device != "gpu",
+            **self.training_args,
+        )
+
+        # The Trainer class is used for fine-tuning the model.
+        trainer = Seq2SeqTrainer(
+            model=self.model,
+            args=training_args,
+            train_dataset=dataset,
+        )
+
+        trainer.train()
+        self.fitted = True
+        shutil.rmtree(
+            "DashAI/back/user_models/temp_checkpoints_opus-mt-en-es", ignore_errors=True
+        )
+        return self
+
+    def predict(self, dataset: DashAIDataset) -> List:
+        """Predicting with the fine-tuned model
+
+        Parameters
+        ----------
+        dataset : DashAIDataset
+            DashAIDataset with text data
+
+        Returns
+        -------
+        List
+            list of translations made by the model
+        """
+        if not self.fitted:
+            raise NotFittedError(
+                f"This {self.__class__.__name__} instance is not fitted yet. Call 'fit'"
+                " with appropriate arguments before using this "
+                "estimator."
+            )
+
+        input_column = dataset.inputs_columns[0]
+
+        def encode(examples):
+            return self.tokenizer(
+                examples[input_column],
+                truncation=True,
+                padding="max_length",
+                max_length=512,
+            )
+
+        dataset = dataset.map(encode, batched=True)
+        dataset.set_format(type="torch", columns=["input_ids", "attention_mask"])
+
+        translations = []
+
+        for example in dataset:
+            inputs = {
+                k: v.unsqueeze(0).to(self.model.device) for k, v in example.items()
+            }
+            outputs = self.model.generate(**inputs)
```

Comment:
Perhaps as a side note, doesn't generate also have its own parameters, such as the type of search to be performed (e.g. use beam search)?
I don't know if you have thought about how to deal with this issue?

Reply:
To be honest I had ignored it. But in that case it could be requested as an extra argument for the model or it could be a separate configurable object in the future like a tokenizer.